options.go

```diff
@@ -91,6 +91,7 @@ func RequestHeader(header string) Option {
 func FieldName(name string) Option {
 	return func(cs *csrf) error {
 		cs.opts.FieldName = name
+		fieldName = name
```

Comment:
This won't work as it pollutes global state—different instances of CSRF will run over each other's configuration by relying on a private package global.

* This function - `csrf.FieldName` needs to stay as it was.
* Add the following line after this line - https://github.com/molivier/csrf/blob/fix/csrf.go#L204-L205 -

```go
cs.c.Env[fieldName] = cs.opts.FieldName
```
* In `csrf.TemplateField` refer to `c.Env["fieldName"]` to get the fieldName from the request context.
* Write a test that overrides the fieldname and checks that the output of `csrf.TemplateField` matches.

Note that `csrf.TemplateField` should be considered a 'convenience' function. Right now (before this PR) you can just write a quick three-line helper that uses your own field name - e.g.

```go
func CSRFField(r *http.Request) template.HTML {
	fragment := fmt.Sprintf(`<input type="hidden" name="%s" value="%s">`,
		myCustomFieldName, Token(c, r))

	return template.HTML(fragment)
}

// Note: you would still set it via csrf.FieldName(myCustomFieldName) - with myCustomField name likely being passed in via your own config file/environment/struct
```